library/kotlin/src/io/envoyproxy/envoymobile/StreamClientBuilder.kt

```diff
@@ -102,6 +103,19 @@ open class StreamClientBuilder(
     return this
   }
 
+  /**
+   * Add an HTTP filter factory used to create filters for streams sent by this client.
+   *
+   * @param filterName: unique name identifying this filter in the chain.
+   * @param factory closure returning an instantiated filter.
+   *
+   * @return this builder.
+   */
+  fun addFilter(filterName: String, factory: () -> Filter): StreamClientBuilder {
```

Comment:
Yeah, I was planning on it. We might be able to replicate it on Kotlin with a crossinline closure with a reified generic, but I was having a good enough time fighting the Kotlin compiler already. And honestly static types on an interface are already a little funky. There are other options... it just has to be unique, so honestly it could even be a guid. But if we want to rely on it for anything (logging? debugging?), human-readable is maybe preferable.

I'll update the Swift interface to match this here anyways, for now.

Reply:
https://github.com/lyft/envoy-mobile/pull/997